### PR TITLE
[QAA-120] Review and optimize e2e testing workflow

### PR DESCRIPTION
**Pull request by ypolishchuk-ledger**
### ✅ Checklist

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

_Replace this text by a clear and concise description of what this pull request is about and why it is needed. Be sure to explain the problem you're addressing and the solution you're proposing._
_For libraries, you can add a code sample of how to use it._
_For bugfixes, you can explain the previous behavior and how it was fixed._
_In case of visual features, please attach screenshots or video recordings to demonstrate the changes._

### ❓ Context

- **JIRA or GitHub link**: 

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: apps/ledger-live-desktop/tests/specs/speculos/add.account.spec.ts [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 import { specs } from "../../utils/speculos";
 import { Currency } from "../../enum/Currency";
-import { Application } from "tests/page";
 import { addTmsLink } from "tests/fixtures/common";
 
 const currencies: Currency[] = [
   Currency.BTC,
   Currency.ETH,
-  Currency.ETC,
+  //Currency.ETC, TODO: enable when bug is merged on develop https://ledgerhq.atlassian.net/browse/LIVE-13049
```
Comment (abdurrahman-ledger) [inline apps/ledger-live-desktop/tests/specs/speculos/add.account.spec.ts:9]: To check if fix was merged to develop

File: apps/ledger-live-desktop/tests/page/install.page.ts [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
-    "data-test-id=debug-install-set-of-apps-button",
-  );
-  private resetButton = this.page.locator("data-test-id=reset-button");
-  private installCtaButton = this.page.locator("data-test-id=install-cta-button");
-  private cancelCtaButton = this.page.locator("data-test-id=skip-cta-button");
-  private inputOptionSelector = this.page.locator("data-test-id=input-option-selector");
+  private debugInstallSetOfAppsButton = this.page.getByTestId("debug-install-set-of-apps-button");
+  private resetButton = this.page.getByTestId("reset-button");
+  private installCtaButton = this.page.getByTestId("install-cta-button");
+  private cancelCtaButton = this.page.getByTestId("kip-cta-button");
```
[the remark this answers is not in the document]
Comment (ypolishchuk-ledger) [reply apps/ledger-live-desktop/tests/page/install.page.ts:7]: indeed, good catch! thank you

File: apps/ledger-live-desktop/tests/page/swap.page.ts [hunk 3]
```diff
@@ -2,54 +2,46 @@ import { waitFor } from "../utils/waitFor";
 import { AppPage } from "tests/page/abstractClasses";
 
 export class SwapPage extends AppPage {
-  private swapMenuButton = this.page.locator("data-test-id=drawer-swap-button"); // TODO: Should this be here?
+  private swapMenuButton = this.page.getByTestId("drawer-swap-button"); // TODO: Should this be here?
```
[the remark this answers is not in the document]
Comment (ypolishchuk-ledger) [reply apps/ledger-live-desktop/tests/page/swap.page.ts:5]: I have no idea, I decided not to remove something which might be needed by other people

File: apps/ledger-live-desktop/tests/page/swap.page.ts [hunk 4]
```diff
@@ -2,54 +2,46 @@ import { waitFor } from "../utils/waitFor";
...
   private currencyByName = (accountName: string) => this.page.getByText(accountName); // TODO: this is rubbish. Changed this
```
[the remark this answers is not in the document]
Comment (ypolishchuk-ledger) [reply apps/ledger-live-desktop/tests/page/swap.page.ts:6]: I have no idea, I decided not to remove something which might be needed by other people

File: apps/ledger-live-desktop/tests/page/swap.page.ts [hunk 5]
```diff
@@ -2,54 +2,46 @@ import { waitFor } from "../utils/waitFor";
...
 
   // Swap Amount and Currency components
```
[the remark this answers is not in the document]
Comment (ypolishchuk-ledger) [reply apps/ledger-live-desktop/tests/page/swap.page.ts:8]: I have no idea, I decided not to remove something which might be needed by other people. In any case this was not the goal of this PR

File: apps/ledger-live-desktop/tests/page/post.onboarding.page.ts [hunk 6]
```diff
@@ -1,34 +1,26 @@
 import { AppPage } from "tests/page/abstractClasses";
 
 export class PostOnboarding extends AppPage {
-  private sideDrawerContainer = this.page.locator("data-test-id=side-drawer-container");
-  private postOnboardingHubTesterButton = this.page.locator(
-    "data-test-id=postonboarding-tester-button",
+  private sideDrawerContainer = this.page.getByTestId("side-drawer-container");
+  private postOnboardingHubTesterButton = this.page.getByTestId("postonboarding-tester-button");
+  readonly postOnboardingHubContainer = this.page.getByTestId("post-onboarding-hub-container");
```
[the remark this answers is not in the document]
Comment (ypolishchuk-ledger) [reply apps/ledger-live-desktop/tests/page/post.onboarding.page.ts:6]: As discussed - better to address this separately

**Review by bharamboure-ledger — DISMISSED** [review]

File: apps/ledger-live-desktop/tests/specs/speculos/add.account.spec.ts (hunk 1 above)
Comment (ypolishchuk-ledger) [reply apps/ledger-live-desktop/tests/specs/speculos/add.account.spec.ts:9]: not fixed, looks like issue is only with speculos

File: .github/workflows/test-ui-e2e-only-desktop.yml [hunk 7]
```diff
@@ -103,55 +100,37 @@ jobs:
...
-            apps/ledger-live-desktop/tests/artifacts/*.log
-      - name: Upload Allure Report
-        if: ${{ !cancelled() }}
-        uses: actions/upload-artifact@v4
+          name: allure-results-${{ matrix.shardIndex }}
+          path: "apps/ledger-live-desktop/allure-results"
+
+  report-and-notify:
+    name: "Report and Notify"
+    runs-on: [ledger-live-4xlarge]
```
[the remark this answers is not in the document]
Comment (ypolishchuk-ledger) [reply .github/workflows/test-ui-e2e-only-desktop.yml]: replaced to smaller one, thanks!

File: .github/workflows/test-ui-e2e-only-desktop.yml [hunk 8]
```diff
@@ -103,55 +100,37 @@ jobs:
...
-        uses: actions/upload-artifact@v4
-        if: failure() && !cancelled()
+        uses: actions/upload-artifact@v4.3.0
         with:
-          name: playwright-results-linux
-          path: |
-            apps/ledger-live-desktop/tests/artifacts/test-results
-            apps/ledger-live-desktop/tests/artifacts/html-report
-            apps/ledger-live-desktop/tests/artifacts/coverage
-            apps/ledger-live-desktop/tests/artifacts/videos
```
[the remark this answers is not in the document]
Comment (ypolishchuk-ledger) [reply .github/workflows/test-ui-e2e-only-desktop.yml:140]: Good point! I brought this step back, and created [this ticket](https://ledgerhq.atlassian.net/browse/QAA-132) to attach video to Allure report

**Review by bharamboure-ledger — DISMISSED** [review]

**Review by valber-ledger — APPROVED** [review]

**Review by abdurrahman-ledger — DISMISSED** [review]

**Review by bharamboure-ledger — APPROVED** [review]

**Review by abdurrahman-ledger — DISMISSED** [review]

**Review by bharamboure-ledger — APPROVED** [review]

**Pull request merged**